src/lib.rs

```diff
@@ -57,6 +70,14 @@ pub enum NumberOrString {
 /// A request that got canceled still needs to return from the server and send a response back.
 /// It can not be left open / hanging. This is in line with the JSON RPC protocol that requires
 /// that every request sends a response back. In addition it allows for returning partial results on cancel.
+#[derive(Debug)]
+pub struct Cancel;
```

Comment:
Since these types aren't meant to be used as values we could have them as uninhabited enums maybe?

```rust
pub enum Cancel { }
```

That way they can only be used in as types  and not values. Or `pub struct Cancel(());` so they can't be instantiated.